local/client/cli/cli.go

```diff
@@ -0,0 +1,45 @@
+package cli
+
+import (
+	"github.com/scootdev/scoot/local/client/conn"
+	"github.com/spf13/cobra"
+)
+
+type CliClient struct {
+	rootCmd *cobra.Command
+	dialer  conn.Dialer
+	conn    conn.Conn
+}
+
+func (c *CliClient) Exec() error {
+	return c.rootCmd.Execute()
+}
+
+func (c *CliClient) Close() error {
+	return c.conn.Close()
+}
+
+func (c *CliClient) openConn() (conn.Conn, error) {
+	if c.conn == nil {
+		conn, err := c.dialer.Dial()
+		if err != nil {
+			return nil, err
+		}
+		c.conn = conn
+	}
+	return c.conn, nil
+}
+
+func NewCliClient(dialer conn.Dialer) (*CliClient, error) {
```

Comment:
This is Dependency Injection. I remember you said you thought sometimes the SOLID principles led to convoluted code.

I think it leads to a nice separation of concerns. The CLI client knows how to translate between command-line and the connection layer; it doesn't know how to create a connection.

Should we talk about how we feel about DI as a team?